tests/lib/dune

```diff
@@ -0,0 +1,10 @@
+(test
+  (name      patcher)
```

Comment:
Patcher move is not directly correlated to this PR, but as it is needed for addition of type gymnastics, to not delay (have a PR from which this one depends) we could keep it.

Reply:
As a general thing, I think having a separate commit is enough - I don't think we'd need to require that to go to a separate PR, unless the reviewing's become too large?